### PR TITLE
Split out capi targets: make separate freertos, staticlib, and cdylib crates as targets

### PR DESCRIPTION
**Pull request by Manishearth**
This allows one to vendor things without pulling in freertos deps, and also is a less weird-looking workaround to https://github.com/rust-lang/cargo/issues/4881

### CODE REVIEW

File: ffi/diplomat/Cargo.toml [hunk 1]
```diff
@@ -47,6 +47,9 @@ provider_static = ["icu_testdata"]
 smaller_static = ["provider_static"]
 # Enables size-optimized builds on x86_64
 x86tiny = ["dlmalloc"]
+# We can't exclude x86tiny from `--all-features` and it messes up the build for other crates
+# Instead, we use this feature to detect when `--all-features` is occurring
+internal_all_features_hack = []
```
Comment (Manishearth) [inline ffi/diplomat/Cargo.toml:52]: We don't really have a good way to opt out of `--all-features`

File: ffi/diplomat/Cargo.toml [hunk 2]
```diff
@@ -31,31 +31,25 @@ all-features = true
...
 # Bench feature gets tested separately and is only relevant for CI.
-# wearos/freertos/x86tiny are not relevant in normal environments,
+# x86tiny is not relevant in normal environments,
 # smaller_static gets tested on the FFI job anyway
-denylist = ["bench", "wearos", "freertos", "x86tiny", "smaller_static"]
+denylist = ["bench", "x86tiny", "smaller_static", "internal_all_features_hack"]
 
 [lib]
-crate-type = ["staticlib", "rlib"]
+crate-type = ["rlib"]
```
Comment (sffc) [inline ffi/diplomat/Cargo.toml:39]: Nit (optional): remove

File: ffi/diplomat/Cargo.toml [hunk 3]
```diff
@@ -31,31 +31,25 @@ all-features = true
 # Omit most optional dependency features from permutation testing
 skip_optional_dependencies = true
 # Bench feature gets tested separately and is only relevant for CI.
-# wearos/freertos/x86tiny are not relevant in normal environments,
+# x86tiny is not relevant in normal environments,
```
Comment (sffc) [inline ffi/diplomat/Cargo.toml:34]: Suggestion: get rid of the `x86tiny` feature in this crate and move it to the crate that needs it (staticlib?)

**Review by sffc — CHANGES_REQUESTED** [review]

File: ffi/capi_cdylib/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    "Cargo.toml",
+    "LICENSE",
+    "README.md"
+]
+
+[package.metadata.docs.rs]
+all-features = true
+
+[lib]
+crate-type = ["cdylib", "rlib"]
```
Comment (sffc) [inline ffi/capi_cdylib/Cargo.toml:27]: Question: do you need the `rlib` here?

File: ffi/capi_staticlib/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    "Cargo.toml",
+    "LICENSE",
+    "README.md"
+]
+
+[package.metadata.docs.rs]
+all-features = true
+
+[lib]
+crate-type = ["staticlib", "rlib"]
```
Comment (sffc) [inline ffi/capi_staticlib/Cargo.toml:27]: Question: do you need the `rlib`?

**Review by sffc — APPROVED** [review]

File: ffi/capi_cdylib/Cargo.toml (hunk 4 above)
Comment (Manishearth) [reply ffi/capi_cdylib/Cargo.toml:31]: sometimes other tools expect it

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**